### PR TITLE
Honeywellabp

### PR DESCRIPTION
**Pull request by RubyBailey**
## Description:

Adding Honeywell ABP sensor (SPI version). I have used this sensor for a number of personal applications. I have appreciated the support from the esphome community in the past and wanted to contribute to esphome.

**Related issue (if applicable):** none

**Pull request in [esphome](https://github.com/esphome/esphome) with YAML changes (if applicable):** esphome/esphome#3164

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

File: components/sensor/honeywellabp.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    # Example configuration entry
+    sensor:
+    - platform: honeywellabp
+        pressure:
+            name: "Honeywell pressure"
+            unit_of_measurement: "psi"
+            min_pressure: 0
+            max_pressure: 15
+        temperature:
+            name: "Honeywell temperature"
```
Comment (OttoWinter) [inline components/sensor/honeywellabp.rst]:
```suggestion
          name: "Honeywell pressure"
          unit_of_measurement: "psi"
          min_pressure: 0
          max_pressure: 15
        temperature:
          name: "Honeywell temperature"
```

We indent with two spaces

File: components/sensor/honeywellabp.rst [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        cs_pin: GPIO5 #D5
+        update_interval: 2s
+
+Configuration variables:
+------------------------
+
+min_pressure and max_pressure are from the device datasheet. These are used to calculate the pressure reading published by the sensor.
```
Comment (OttoWinter) [inline components/sensor/honeywellabp.rst]:
```suggestion
The values for ``min_pressure`` and ``max_pressure`` can be found in the datasheet for the specific device These are used to calculate the pressure reading published by the sensor.
```

File: components/sensor/honeywellabp.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+- **pressure** (*Optional*): The information for the pressure sensor.
+
+  - **name** (**Required**, string): The name for the pressure sensor.
+  - **min_pressure** (**Required**, int or float): Minumim pressure for the pressure sensor.
```
Comment (OttoWinter) [inline components/sensor/honeywellabp.rst]: What units are these in? We should put that information here

```suggestion
  - **min_pressure** (**Required**, int or float): Minimum pressure for the pressure sensor.
```

File: components/sensor/honeywellabp.rst [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  - **max_pressure** (**Required**, int or float): Maximum pressure for the pressure sensor.
+  - **id** (*Optional*, :ref:`config-id`): Set the ID of this sensor for use in lambdas.
+  - **unit_of_measurement** (*Optional*, string): Defaults to ``psi``
```
Comment (OttoWinter) [inline components/sensor/honeywellabp.rst]:
```suggestion
```

Ah I see so the above is in psi. Putting it here makes it more confusing though IMO, can you just add it above like `Minimum pressure for the pressure sensor in psi`.

File: components/sensor/honeywellabp.rst (hunk 1 above)
Comment (RubyBailey) [reply components/sensor/honeywellabp.rst]: Done. thanks.

File: components/sensor/honeywellabp.rst (hunk 4 above)
Comment (RubyBailey) [reply components/sensor/honeywellabp.rst]: I added some text clarifying the unit of measurements and the default units for minimum and maximum pressure. Hope that makes things more clear. Thanks again for the comments.

**Pull request merged**